src/dfcx_scrapi/core/security_settings.py

```diff
@@ -0,0 +1,215 @@
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import json
+import logging
+from google.cloud.dialogflowcx_v3beta1 import services
+from google.cloud.dialogflowcx_v3beta1 import types
+from google.protobuf import field_mask_pb2
+from dfcx_scrapi.core.scrapi_base import ScrapiBase
+
+
+# logging config
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)-8s %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+class SecuritySettings(ScrapiBase):
+    """Core Class for CCAI Security Settings."""
+
+    def __init__(
+        self,
+        creds_path: str = None,
+        creds_dict: dict = None,
+        creds=None,
+        scope=False,
+        agent_id: str = None
+    ):
+        super().__init__(
+            creds_path=creds_path,
+            creds_dict=creds_dict,
+            creds=creds,
+            scope=scope,
+        )
+
+        if agent_id:
+            self.agent_id = agent_id
+
+        self.ss_service = services.security_settings_service
+        self.ss_types = types.security_settings
+
+    def list_security_settings(self, location_id: str):
+        """List Security Settings for a given Project and Region.
+
+        Args:
+            location_id: CX Agent ID string in the following format
+            projects/<PROJECT ID>/locations/<LOCATION ID>/
+
+        Returns:
+            results: Dict of results
+        """
+
+        request = self.ss_types.ListSecuritySettingsRequest()
+
+        request.parent = location_id
+        client_options = self._set_region(location_id)
+        client = self.ss_service.SecuritySettingsServiceClient(
+            credentials=self.creds, client_options=client_options
+        )
+
+        response = client.list_security_settings(request)
+
+        security_settings = []
+        for page in response.pages:
+            for ss in page.security_settings:
+                security_settings.append(ss)
+
+        return security_settings
+
+
+    def get_security_settings(self, security_setting_id: str):
+        """Get specified CCAI Security Setting.
+
+        Args:
+            security_setting_id: Security Setting ID string in the following
+                format: projects/<PROJECT ID>/locations/<LOCATION ID>/
+                securitySettings/<SECURITY SETTINGS ID>
+
+        Returns:
+            response: a single Security Settings object of
```

Comment:
is it normal to provide a variable name for the return value? I've seen this a lot and it doesn't make much sense to me

Reply:
@Greenford yeah I suppose this is unnecessary. I can clean this up in this file and then we can start cleaning it up across the other files as we make updates to them as well.